packages/mobx-state-tree/src/core/mst-operations.ts

```diff
@@ -647,6 +647,38 @@ export function walk(target: IStateTreeNode, processor: (item: IStateTreeNode) =
     processor(node.storedValue)
 }
 
+export interface IReflected {
+    name: string
+    properties: { [K: string]: IType<any, any> }
+    actions: string[]
+    views: string[]
+    volatile: string[]
+}
+/**
+ * Returns a reflection of the node
+ *
+ * @export
+ * @param {IStateTreeNode} target
+ * @returns {IReflected}
+ */
+export function reflect(target: IStateTreeNode): IReflected {
+    // check all arguments
+    if (process.env.NODE_ENV !== "production") {
+        if (!isStateTreeNode(target))
```

Comment:
I think we should also check if it is a model type specifically? (not array or map)

Reply:
currently it also works with a map - that's also why the subtype

Reply:
Never mind, you made that work as well :)